forge-game/src/main/java/forge/game/card/Card.java

```diff
@@ -4669,9 +4680,20 @@ public final SpellAbility getSpellAbilityForStaticAbility(final String str, fina
     }
 
     public final Trigger getTriggerForStaticAbility(final String str, final StaticAbility stAb) {
+        Map<String, String> changedTextColors = stAb.getChangedTextColors();
+        Map<String, String> changedTextTypes = stAb.getChangedTextTypes();
         Trigger result = storedTrigger.get(stAb, str);
+        if (result != null) {
+            boolean needNewCache = !result.getChangedTextColors().equals(changedTextColors);
```

Comment:
if the Text of the static host gets changed *again* (after it's cached) we might be stuck with the previous version. Is there a better way compared to these checks?

Reply:
move this check to CardTraitsBase

like "sameChangedText(CardTraitBase other)"

Reply:
i might have moved to card traits, but this is fine too